### ISSUE TITLE
Work out how to handle multiple pages

### ISSUE THREAD

**Issue by ajyoon** [ajyoon/brown#10]
Multiple paged documents are currently handled in a very broken, hacky way in `FlowableFrame` and `Document`. The API doesn't properly support multiple page documents at all. There should be a consistent, general way to deal with paging.

A likely option is to extend `Point` with a `page` property:

```python
Point(x=0, y=0, page=1)
```

The `page` property could be a simple `int` that defaults to `1` (or `0`?) `GraphicObjects` which deal with `Point` positions will treat `page` as a relative delta, just as `x` and `y` are. 

By default (and in `GraphicObject`s whose parent is the document), the `x` and `y` coordinates would be taken as relative to the live page area - the top left corner of the page margins.

A possible point of confusion / complication is that the typical (in publication language) page 1 would be `page=0` in this system. This would be necessary to maintain the integrity of relative points. For instance, in this new system, it would be expected that the following objects would be positioned at the same global location:

```python
some_parent = GraphicObject(Point(Mm(42), Mm(71), page=4), parent=brown.document)
some_child = GraphicObject(Point(Mm(0), Mm(0), page=0), parent=some_parent)
```

**Comment by ajyoon** [comment]
Groundwork simplifying `Point` init signature in b80960fbfa2b6c48196f646bd1548356d3472aad necessary to extend the `Point` constructor pattern cleanly.

**Comment by ajyoon** [comment]
More work simplifying `Point` and `AnchoredPoint` in 7ac3a86974df7b7f5da50458ecd1ffb87a1c265e and 381265e2bb6cd1c92c1bbe613f6ed62795b95157

**Comment by ajyoon** [comment]
`Point.page` and `AnchoredPoint.page` implemented in 1ed2a825779b2b50ce479dd327441cc729ad02cb. This has no effect on rendering yet, but most of the groundwork on the non-rendering side of this is now in place.

**Comment by ajyoon** [comment]
As part of simplifying the `Point` API across the board, moving toward removing `AnchoredPoint` entirely. d073234e0a1373682a2d873593cec78356a08051 removes it from `Path`

**Comment by ajyoon** [comment]
Implemented in ed10238b6abc21686cfdd41998b5b6f04cf12415

From commit message:
```
This fixes the broken, inconsistent way of handling multi-page
documents by having GraphicObjectInterface compute the canvas
positions of all coordinates by using the Point.page attribute
in conjunction with new logic in Document. All non-interface
API's now deal exlusively with logical (page + position on page)
coordinates. The transformation to canvas positions occurs
at the step where interface classes pass coordinates to Qt.
```
